### PR TITLE
Passing next to response handler, so request can be closed.

### PR DESCRIPTION
**Pull request by manokovacs**
New feature of responseHandler is nice but faulty.

### CODE REVIEW

**Comment by siacomuzzi** [comment]
* please use the following signature: `responseHandler(SAMLRequest, req, res, next);`
* also update the readme

**Comment by manokovacs** [comment]
Changes added as requested.

File: lib/samlp.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ var algorithms = {
  * - cert the public certificate
  * - key the private certificate to sign all tokens
  * - postUrl function (SAMLRequest, request, callback)
- * - responseHandler(request, response, SAMLRequest) a function that handles the response. Defaults to HTML POST to postUrl.
+ * - responseHandler(SAMLRequest, request, response, next) a function that handles the response. Defaults to HTML POST to postUrl.
```
Comment (siacomuzzi) [inline lib/samlp.js]: `responseHandler(SAMLResponse, request, response, next)`

File: lib/samlp.js [hunk 2]
```diff
@@ -225,6 +225,7 @@ module.exports.auth = function(options) {
           RelayState:      opts.RelayState || (req.query || {}).RelayState || (req.body || {}).RelayState || '',
           SAMLResponse:    response.toString('base64')
         }));
+        next();
```
Comment (siacomuzzi) [inline lib/samlp.js]: is it necessary?

**Comment by manokovacs** [comment]
When running these in async integration tests, it is necessary to receive the end of a controller execution.

File: lib/samlp.js (hunk 2 above)
Comment (siacomuzzi) [inline lib/samlp.js]: if you call `res.send()`, it calls `res.end()` for you, so you could override it:

```
var end = res.end;
res.end = function(chunk, encoding) {
  // your asserts

  res.end = end;
  res.end(chunk, encoding);
};
```

Comment (manokovacs) [inline lib/samlp.js]: Removed.

**Comment by siacomuzzi** [comment]
before merge it, please check the following:
* https://github.com/auth0/node-samlp/pull/15/files#r43134754
* squash PR commits into one: https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

**Comment by manokovacs** [comment]
Squash pushed!

**Comment by siacomuzzi** [comment]
`samlp@0.13.0`

thanks @manokovacs ! :+1:

**Pull request merged**